src/middlewares/authMiddleware.js

```
@@ -0,0 +1,25 @@
+const jwt = require('jsonwebtoken');
+const JWT_SECRET = process.env.JWT_SECRET;
+
+const authenticate = (req, res, next) => {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader || !authHeader.startsWith('Bearer ')) {
+    res.status(401).json({ message: 'No token provided' });
+
+    return;
+  }
+
+  const token = authHeader.split(' ')[1];
+
+  try {
+    const decoded = jwt.verify(token, JWT_SECRET);
+
+    req.user = decoded;
+    next();
+  } catch (err) {
+    res.status(401).json({ message: 'Invalid or expired token' });
```

In the try/catch you verify the token and set `req.user = decoded`. The catch block returns 401 but swallows the underlying error. Change `catch (err)` and log `err` (console.error or a logger) so expired/malformed token reasons are visible in server logs. This aligns with previous guidance to avoid silent failures and improves debuggability across the app.